libraries/anvill_passes/src/RecoverStackFrameInformation.cpp

```diff
@@ -265,34 +266,32 @@ RecoverStackFrameInformation::UpdateFunction(
   // IRBuilder, and then create an `alloca` instruction to
   // generate our new stack frame
   auto &entry_block = function.getEntryBlock();
+  auto &insert_point = *entry_block.getFirstInsertionPt();
 
-  auto first_instr_it = entry_block.begin();
-  if (first_instr_it == entry_block.end()) {
-    return StackAnalysisErrorCode::InternalError;
-  }
-
-  auto first_instr = &(*first_instr_it);
-
-  llvm::IRBuilder<> builder(first_instr);
+  llvm::IRBuilder<> builder(&insert_point);
   auto stack_frame_alloca = builder.CreateAlloca(stack_frame_type);
 
   // Pre-initialize the stack frame to zero if we have been requested
   // to do so. From the whole FunctionPass class, we get the setting
   // from the LiftingOptions class
-  if (initialize_stack_frame) {
-    for (auto i = 0U; i < stack_frame_analysis.size; ++i) {
-      auto stack_frame_byte = builder.CreateGEP(
-          stack_frame_alloca,
-          {builder.getInt32(0), builder.getInt32(0), builder.getInt32(i)});
+  if (init_strategy == StackFrameStructureInitializationProcedure::kZeroes) {
```

Comment:
Use a `switch` on `init_strategy`, so adding new cases triggers a compiler warning/error for unhandled cases.